src/lib/__tests__/InstantSearch-test.js

```diff
@@ -50,6 +50,29 @@ See: https://www.algolia.com/doc/guides/building-search-ui/going-further/backend
 `);
   });
 
+  it('throws with invalid insightsClient', () => {
+    expect(() => {
+      // eslint-disable-next-line no-new
+      new InstantSearch({
+        indexName: 'indexName',
+        searchClient: { search: () => {} },
+        insightsClient: 'insights',
+      });
+    }).toThrowErrorMatchingInlineSnapshot(
+      `"The provided \`insightsClient\` must be a function."`
+    );
+  });
+
+  it('stores insightsClient on instance', () => {
```

Comment:
This is an implementation detail, right?

I don't think we need this test, especially because we don't document this usage.

Reply:
connectInsights expects it to be here, so it's part of the contract this module is making with other modules of the code. 
if you change something that break this contract, you need to be able to detect it. This test is here for that.
I still think it makes sense to keep it.

Reply:
Right, if other pieces of the codebase depend on it, it makes sense to test it.